tests/test_grids.py

```diff
@@ -675,8 +675,10 @@ def test_mitgridindexing_3D(mode, gridindexingtype, withtime):
         dimensions = {"lon": lon, "lat": lat, "depth": depth}
         dsize = (depth.size, lat.size, lon.size)
 
-    index_signs = {'nemo': -1, 'mitgcm': 1}
-    isign = index_signs[gridindexingtype]
+    hindex_signs = {'nemo': -1, 'mitgcm': 1, 'mom5': -1}
```

Comment:
I've been trying to wrap my head around it for a while, but also not yet sure why this passes. I'll look at it again next week, with a clear mind.

If MOM5s indexing is different than that in POP, we should have a unit test for both. Once we have the sample data from @hrsdawson, we'll also better be able to compare differences in both model outputs.

Reply:
OK, I think I know why changing `vindex_signs` does not make any difference to the particle trajectory. It's because the `W` field for such a radial flow does not vary with depth, only with longitude. So no matter how we change `depth[k] + vsign * dz` in calculating `r` and `phi`, `W` will stay the same.

So the question then is: is this a non-useful idealised unit test for this mom-grid example? Or am I thinking wrongly about how to calculate the grid levels for `U` and `W`? @daanreijnders and @hrsdawson, what are your thoughts?

Reply:
Why does W only change with longitude only change with longitude and not with depth? Particles are advected clockwise on a circle in the XZ-plane, right? W then changes with depth to provide the up- and downward motion to keep it on the circle.

Reply:
This should be resolved with https://github.com/OceanParcels/parcels/pull/923/commits/062ce93a91daf0b3c78560a30ac57cc57987726e. The problem was indeed that W in its current definition had no vertical gradient, so both indexing types (`mom5` and `pop`-like B-grids) would sample the same velocities. The solution was to rotate the coordinate system (swap U and W), also in the kernel.